rdock-utils/rdock_utils/sdfilter/filter.py

```diff
@@ -0,0 +1,105 @@
+# Standard Library
+import logging
+import operator
+import shlex
+from collections import Counter
+from pathlib import Path
+from typing import Any, Callable, Iterable
+
+# Local imports
+from rdock_utils.common.SDFParser import FastSDMol
+
+logger = logging.getLogger("sdfilter")
+
+
+def get_casted_operands(operand1: str | int, operand2: str | int) -> tuple[str, str] | tuple[float, float]:
+    try:
+        return (float(operand1), float(operand2))
+    except ValueError:
+        return (str(operand1), str(operand2))
+
+
+OPERATION_MAP: dict[str, Callable[[Any, Any], bool]] = {
+    "==": operator.eq,
+    "!=": operator.ne,
+    "<": operator.lt,
+    ">": operator.gt,
+    "<=": operator.le,
+    ">=": operator.ge,
+    "eq": operator.eq,
+    "ne": operator.ne,
+    "lt": operator.lt,
+    "gt": operator.gt,
+    "le": operator.le,
+    "ge": operator.ge,
+    "in": lambda x, y: x in y,
+    "not_in": lambda x, y: x not in y,
```

Comment:
Consider using 'operator.contains' and 'not operator.contains' which are equivalent to "in" and "not_in" operations

Reply:
the problem with those is the order of operators is reversed:
```
operator.contains(x, y) == y in x
```
this would require either special cases for `in` and `not in` operators, or changing the 'current' behavior of the script

I thought about writing this in a comment, as it feels a bit annoying having every other operator coming from `operator` module, and `in` and `not in` being a lambda, and it looks like it would have been useful :sweat_smile:

Reply:
True..., I didn't realize this cannot be implemented with the current behavior :sweat_smile: